subsys/bluetooth/controller/ll_sw/nordic/lll/lll_clock.c

```diff
@@ -66,6 +66,9 @@ int lll_clock_deinit(void)
 	struct onoff_manager *mgr =
 		z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_LF);
 
+	/* Cancel any ongoing request */
+	(void)onoff_cancel(mgr, &lf_cli);
+
```

Comment:
Are you sure cancel does not make the release to fail?

```
/**
* @brief Attempt to cancel an in-progress client operation.
*
* It may be that a client has initiated an operation but needs to
* shut down before the operation has completed.  For example, when a
* request was made and the need is no longer present.
*
* Cancelling is supported only for onoff_request() and onoff_reset()
* operations, and is a synchronous operation.  Be aware that any
* transition that was initiated on behalf of the client will continue
* to progress to completion: it is only notification of transition
* completion that may be eliminated.  If there are no active requests
* when a transition to on completes the manager will initiate a
* transition to off.
*
* Client notification does not occur for cancelled operations.
 
```

I feel the onoff module needs a fix... both cancel and release are synchronous call by documentation. The release should take care of any pending requests.

```
/**
* @brief Release a reserved use of an on-off service.
*
* This synchronously releases the caller's previous request.  If the
* last request is released the manager will initiate a transition to
* off, which can be observed by registering an onoff_monitor.
*
* @note Behavior is undefined if this is not paired with a preceding
* onoff_request() call that completed successfully.
*
* @param mgr the manager for which a request was successful.
*
```

Reply:
I tested it with the linked PR, and it works

Reply:
@andyross and/or @nashif  as you are listed as maintainer of onoff module, what is the correct usage of `onoff_release()`? a prior `onoff_cancel()`  is always needed?  The documentation suggests that `onoff_release()` is a synchronous call, hence is `onoff_cancel()` really needed before a call to `onoff_release()` ?

Reply:
Here is the failing line: https://github.com/zephyrproject-rtos/zephyr/blob/main/lib/utils/onoff.c#L490

Reply:
@cvinayak Heh, onoff is a Peter Bigot thing, I suspect Anas and I just inherited it by default.   It's a IMHO sort of horrifyingly huge state machine built around one bit (!) of state that probably should have just been a semaphore to begin with.

The code is pretty simple though, and as I read it you **do** need to cancel an active client.  If reset() is called, it's going to do an unconditional sys_slist_append() of that client struct, which is going to break any lists in which it's already present.

Really though: my strong recommendation is to avoid this API and to use a better supported IPC tool.  Though that may not be in scope for this PR and my recommendations in the bluetooth subsystem shouldn't carry much weight.

Reply:
The Controller accepted the use of onoff as a contribution and did not bother as long as it worked, that is, until the necessity to be able to disable Bluetooth subsystem as runtime.

> my strong recommendation is to avoid this API and to use a better supported IPC tool. 

Noted, as the Controller mostly uses the Clock Driver from ISRs, maybe will look in the future for a lightweight alternative.